sdk/tests/js/tests/tex-image-and-sub-image-3d-with-image-bitmap-from-image-bitmap.js

```diff
@@ -0,0 +1,169 @@
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+function generateTest(internalFormat, pixelFormat, pixelType, prologue, resourcePath) {
+    var wtu = WebGLTestUtils;
+    var tiu = TexImageUtils;
+    var gl = null;
+    var successfullyParsed = false;
+    var blackColor = [0, 0, 0];
+    var redColor = [255, 0, 0];
+    var greenColor = [0, 255, 0];
+    var bitmaps = [];
+
+    function init()
+    {
+        description('Verify texImage3D and texSubImage3D code paths taking ImageBitmap created from an ImageBitmap (' + internalFormat + '/' + pixelFormat + '/' + pixelType + ')');
+
+        if(!window.createImageBitmap || !window.ImageBitmap) {
+            finishTest();
+            return;
+        }
+
+        gl = wtu.create3DContext("example");
+
+        if (!prologue(gl)) {
+            finishTest();
+            return;
+        }
+
+        switch (gl[pixelFormat]) {
+          case gl.RED:
+          case gl.RED_INTEGER:
+            greenColor = [0, 0, 0];
+            break;
+          default:
+            break;
+        }
+
+        gl.clearColor(0,0,0,1);
+        gl.clearDepth(1);
+        gl.disable(gl.BLEND);
+
+        var imageData = new ImageData(new Uint8ClampedArray(
+                                      [255, 0, 0, 255,
+                                      255, 0, 0, 0,
+                                      0, 255, 0, 255,
+                                      0, 255, 0, 0]),
+                                      2, 2);
+
+        var bitmap1; // bitmap1 will be in premultiplied format
+        var bitmap2; // bitmap2 will be in unpremultiplied format
+        var p1 = createImageBitmap(imageData).then(function(imageBitmap) { bitmap1 = imageBitmap });
+        var p2 = createImageBitmap(imageData, {imageOrientation: "none", premultiplyAlpha: "none"}).then(function(imageBitmap) { bitmap2 = imageBitmap });
+
+        Promise.all([p1, p2]).then(function() {
+            // Now create new ImageBitmaps from the above two ImageBitmaps.
+            var p3 = createImageBitmap(bitmap1).then(function(imageBitmap) { bitmaps.defaultOption1 = imageBitmap });
+            var p4 = createImageBitmap(bitmap1, {imageOrientation: "none", premultiplyAlpha: "default"}).then(function(imageBitmap) { bitmaps.noFlipYPremul1 = imageBitmap });
+            var p5 = createImageBitmap(bitmap1, {imageOrientation: "none", premultiplyAlpha: "none"}).then(function(imageBitmap) { bitmaps.noFlipYUnpremul1 = imageBitmap });
+            var p6 = createImageBitmap(bitmap1, {imageOrientation: "flipY", premultiplyAlpha: "default"}).then(function(imageBitmap) { bitmaps.flipYPremul1 = imageBitmap });
+            var p7 = createImageBitmap(bitmap1, {imageOrientation: "flipY", premultiplyAlpha: "none"}).then(function(imageBitmap) { bitmaps.flipYUnpremul1 = imageBitmap });
+            var p8 = createImageBitmap(bitmap2).then(function(imageBitmap) { bitmaps.defaultOption2 = imageBitmap });
```

Comment:
Here again I am dubious of the value of testing the default options for the constructor arguments, in particular premultiplyAlpha="default", since its behavior is allowed to vary across browsers.